### DISCUSSION TITLE
Uncaught N-API callback exception detected aka Bad MIDI Value error

### DISCUSSION THREAD

**Discussion by tobyhede** [djipco/webmidi#338] [Q&A] [answered]
I keep encountering some strange exceptions when playing notes.

Have a very basic note on/off setup:
```
        channel.sendNoteOn('C3', {attack: 0.5} )
        channel.sendNoteOff('C3');
```
The MIDI messages are being sent fine. 

However, Node was nagging me with a message saying
```
(node:20096) [DEP0168] DeprecationWarning: Uncaught N-API callback exception detected, please run node with option --force-node-api-uncaught-exceptions-policy=trueto handle those exceptions properly.
```

When I use `-force-node-api-uncaught-exceptions-policy=trueto ` I can see the exception details inside JZZ. 

```
   RangeError: Bad MIDI value: 144,48,64

      at _throw (node_modules/jzz/javascript/JZZ.js:1635:30)
      at MIDI (node_modules/jzz/javascript/JZZ.js:1550:49)
      at Object.handle (node_modules/jzz/javascript/JZZ.js:832:25)
      at node_modules/jzz/javascript/JZZ.js:837:66
```

Obviously this might be something inside JZZ, but I was wondering if anyone had seen something like this before. 
Weird thing is that MIDI messages are still sent, and there seen to be more errors than initial send events.
The MIDI values correspond to the Note and Velocity I am sending.

**Comment by djipco** [comment, marked as answer]
I too started seeing that a while ago. It seems the Node.js team deprecated something that's being used in the `jzz` module.  We probably should report that to the [Jazz-Soft team](https://github.com/jazz-soft/JZZ).

There is [an issue](https://github.com/nodejs/node/issues/15371) related to that on the Node.js GitHub.

**Comment by tobyhede** [comment]
Thanks @djipco - good to confirm that it is not just me. 
Will raise as an issue with JZZ.js.

**Comment by mimimind** [comment]
there may have a way to solve .when in \libsvm-3.22\tools run **python grid.py train_data.txt**  at cmd. it might work or not
